toolkit/analysis_and_visualization/classification/assets/scripts/pagination.js

```diff
@@ -0,0 +1,1125 @@
+                var formatAjaxParams = {
+                    type: "get",
+                    cache: false,
+                    data: {},
+                    contentType: "application/x-www-form-urlencoded; charset=UTF-8",
+                    dataType: "json",
+                    async: true
+                };
+
+                $.extend(true, formatAjaxParams, ajaxParams);
+                $.extend(formatAjaxParams.data, postData);
+
+                formatAjaxParams.url = attributes.dataSource;
+                formatAjaxParams.success = function (response) {
+                    if (self.isDynamicTotalNumber) {
+                        self.findTotalNumberFromRemoteResponse(response);
+                    } else {
+                        self.model.totalNumber = attributes.totalNumber;
+                    }
+
+                    var finalData = self.filterDataByLocator(response);
+                    render(finalData);
+                };
+                formatAjaxParams.error = function (jqXHR, textStatus, errorThrown) {
+                    attributes.formatAjaxError && attributes.formatAjaxError(jqXHR, textStatus, errorThrown);
+                    self.enable();
+                };
+
+                self.disable();
+
+                $.ajax(formatAjaxParams);
+
+                function render(data) {
+                    // Will be invoked before paging
+                    if (self.callHook("beforePaging", pageNumber) === false) return false;
+
+                    // Pagination direction
+                    model.direction = typeof model.pageNumber === "undefined" ? 0 : (pageNumber > model.pageNumber ? 1 : -1);
+
+                    model.pageNumber = pageNumber;
+
+                    self.render();
+
+                    if (self.disabled && self.isAsync) {
+                        // enable pagination
+                        self.enable();
+                    }
+
+                    // cache model data
+                    container.data("pagination").model = model;
+
+                    // format result data before callback invoked
+                    if (attributes.formatResult) {
+                        var cloneData = $.extend(true, [], data);
+                        if (!Helpers.isArray(data = attributes.formatResult(cloneData))) {
+                            data = cloneData;
+                        }
+                    }
+
+                    container.data("pagination").currentPageData = data;
+
+                    // invoke callback
+                    self.doCallback(data, callback);
+
+                    self.callHook("afterPaging", pageNumber);
+
+                    // pageNumber now is the first page
+                    if (pageNumber == 1) {
+                        self.callHook("afterIsFirstPage");
+                    }
+
+                    // pageNumber now is the last page
+                    if (pageNumber == self.getTotalPage()) {
```

Comment:
Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6397592)